src/main/java/macrobase/analysis/classify/OutlierClassifier.java

```diff
@@ -1,27 +1,14 @@
 package macrobase.analysis.classify;
 
+import macrobase.analysis.pipeline.operator.MBConsumer;
+import macrobase.analysis.pipeline.operator.MBProducer;
 import macrobase.analysis.result.OutlierClassificationResult;
-import macrobase.conf.MacroBaseConf;
 import macrobase.datamodel.Datum;
 
-import java.util.Iterator;
-
 /**
  * Consumes a tuple at-a-time from a FeatureTransform and for each tuple outputs
  * a OutlierClassificationResult that should indicate if the tuple observed was an
  * outlier or not.
  */
-public abstract class OutlierClassifier implements Iterator<OutlierClassificationResult> {
-
-    protected Iterator<Datum> input;
-
-    public OutlierClassifier(MacroBaseConf conf, Iterator<Datum> input) {
-        this.input = input;
-    }
-
-    @Override
-    public abstract boolean hasNext();
-
-    @Override
-    public abstract OutlierClassificationResult next();
+public abstract class OutlierClassifier implements MBConsumer<Datum>, MBProducer<OutlierClassificationResult> {
```

Comment:
What do you think about changing OutlierClassifier, FeatureTransform, and Summarizer all to interfaces? That way it makes it easy to implement any shims for testing without implementing all of the functionality a real classifier. Basically we can decouple code reuse from type definitions.